### PR TITLE
feat: always print summary after tests (like SimpleCov)

### PR DESCRIPTION
**Pull request by pftg**
## Summary
- Add `#summary` method to HTML reporter that prints after every test run
- Follows SimpleCov pattern: always show status so devs know reporter is active

## Output examples
```
[snap_diff] 47 screenshots compared, 2 failures. Report: doc/screenshots/snap_diff_report.html
[snap_diff] 47 screenshots compared, no failures.
```

When no screenshots are taken (reporter not used or disabled), nothing prints.

## Context
In jetthoughts.github.io, after upgrading to v1.12.0 with the HTML reporter enabled, there was no visible output confirming the reporter was working. SimpleCov always prints `Coverage report generated for ... to coverage/index.html` regardless of pass/fail.

## Test plan
- [x] 5 new tests for summary output (failures, pass, empty)
- [x] 223 unit tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add a summary output for the HTML reporter that always prints a concise test run status at exit.

New Features:
- Introduce an HTML reporter summary method that reports screenshot counts, failure status, and report path.
- Print a standardized snap_diff summary line after each run when the reporter is used.

Tests:
- Add unit tests covering HTML reporter finalize return values and summary output for failing, passing, and empty runs.

### CODE REVIEW

File: test/unit/reporters/html_reporter_test.rb [hunk 1]
```diff
@@ -146,6 +146,49 @@ def synchronize
...
+      test "#summary returns screenshot count and status" do
+        reporter = HTML.new(output_path: @output_path)
+        reporter.record([build_passing_assertion("ok"), build_failing_assertion("fail")])
+        reporter.finalize
+
+        summary = reporter.summary
+        assert_includes summary, "1 failure"
+        assert_includes summary, "2 screenshots"
+        assert_includes summary, @output_path.to_s
+      end
```
[the remark this answers is not in the document]
Comment (pftg) [reply test/unit/reporters/html_reporter_test.rb:174]: Already covered — test at line 176 \`"#summary pluralizes failures label for multiple failures"\` asserts \`"2 failures"\` and \`"2 screenshots"\`.
